db/migrate/20180907121447_create_timeline_entries.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateTimelineEntries < ActiveRecord::Migration[5.2]
+  def change
+    create_table :timeline_entries do |t|
+      t.string :entry_type, null: true
+      t.references :document, foreign_key: true, null: true
```

Comment:
Should we allow this to be null? I see we have `dependant: destroy` (although not sure if that's just the linter) but we could just have this as a foreign_key with `on_delete: cascade` to not worry about Ruby having to do anything if a document is deleted.

Reply:
Good point @kevindew - not sure why the 'entry_type' one should be null either.

Reply:
🤕 Ouch this should have been `null: false`, misremembered the syntax

Reply:
Updated this to add `null: false` and constraints, including comments with the intention of the constraints. Rubocop still wants me to add `dependent: :destroy` though :|